multimedia/mlt/Portfile

```diff
@@ -80,100 +77,54 @@ depends_lib-append  port:atk \
-                    --disable-swfdec \
-                    --disable-opencv \
-                    --disable-sdl \
-                    --without-kde
+                    -DGPL=ON \
+                    -DGPL3=OFF \
+                    -DMOD_JACKRACK=OFF \
+                    -DMOD_OPENCV=OFF \
+                    -DMOD_SDL1=OFF \
+                    -DMOD_QT=OFF \
+                    -DMOD_QT6=OFF
 
 if {[string match *clang* ${configure.cxx}] && ${configure.cxx_stdlib} ne ""} {
     configure.ldflags-append -stdlib=${configure.cxx_stdlib}
 }
 
-pre-configure {
-    configure.env-append \
-                    QTDIR=${qt_dir}
-    # NB: qt-includedir can and should be set to ${qt_includes_dir},
-    # not ${qt_includes_dir}/Qt !!
-    configure.args-append \
-                    --qt-libdir=${qt_frameworks_dir} \
-                    --qt-includedir=${qt_includes_dir}
-
-    configure.cflags-append  "-I${qt_includes_dir}"
-}
-
-platform darwin 10 {
-    if {${build_arch} eq "x86_64"} {
-        configure.args-delete --disable-mmx
-    }
-}
-
-configure.cflags-append     "-I${prefix}/include"
-configure.ldflags-delete    "-L${prefix}/lib"
-
-# make sure the build system actually sees our settings by amending the toplevel config.mak.
-post-configure {
-    set config.mak [open ${worksrcpath}/config.mak "a"]
-    puts ${config.mak} "CC=${configure.cc}"
-    puts ${config.mak} "CXX=${configure.cxx}"
-    puts ${config.mak} "CFLAGS+=${configure.cflags} ${configure.cppflags}"
-    puts ${config.mak} "CXXFLAGS+=${configure.cxxflags} ${configure.cppflags}"
-    puts ${config.mak} "LDLAGS+=${configure.ldflags}"
-    close ${config.mak}
-}
-
-build.args-append           CC="${configure.cc}" CXX="${configure.cxx}" CPP="${configure.cpp}"
-
-if {${universal_possible} && [variant_isset universal]} {
-    foreach arch ${configure.universal_archs} {
-        lappend merger_build_env(${arch})   "CFLAGS=${configure.cflags} -arch ${arch}"
-        lappend merger_build_env(${arch})   "CXXFLAGS=${configure.cxxflags} -arch ${arch}"
-        lappend merger_build_env(${arch})   "LDFLAGS=${configure.ldflags} -L${worksrcpath}-${arch}/src/framework -L${prefix}/lib -arch ${arch}"
-    }
-} else {
-    build.env-append                        "CFLAGS=${configure.cflags} [get_canonical_archflags cc]"
-    build.env-append                        "CXXFLAGS=${configure.cxxflags} [get_canonical_archflags cxx]"
-    build.env-append                        "LDFLAGS=${configure.ldflags} -L${worksrcpath}/src/framework -L${prefix}/lib [get_canonical_archflags ld]"
-}
-
-variant gtk2 description {include the gtk2 module (for additional image and text producers)} {
-    configure.args-delete   --disable-gtk2
-    depends_lib-append      path:lib/pkgconfig/gtk+-2.0.pc:gtk2
-    require_active_variants gtk2 x11
-}
-
 variant gpl3 description {enable GPLv3 components} {
-    configure.args-append   --enable-gpl3
+    configure.args-replace  -DGPL3=OFF -DGPL3=ON
```

Comment:
Do you think this variant should be a default? I do not know Macports convention in this regard, but it seems a lot people will want more than less.

Reply:
I'm not super familiar with the port tbh, it's not clear to me what components this enables so I don't have a view on it.

Reply:
Taking a look at the mlt code, it seems to be used exclusively for the vqm transition. I think it's a bit strange that GPL and GPL3 are enabled by default in CMakeLists.txt given that the first thing the README mentions is that "MLT is a LGPL multimedia framework". They really should be opt in.

Reply:
The framework is LGPL but not most of the modules. I am not changing the defaults in MLT based on your opinion.

Reply:
Per the CMake I see 27 modules and 8 of them GPL.

Reply:
I changed my comment "most" to "not all" for what it's worth.

Reply:
Keeping the variant as it is for now.